dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -547,6 +549,23 @@ private String deleteDraftFiles(List<String> solrDocIdsForDraftFilesToDelete) {
         return indexResponse.toString();
     }
 
+    private Date findReleaseDate(IndexableDataset indexableDataset) {
+        DatasetVersion currentVersion = indexableDataset.getDatasetVersion();
+        Dataset dataset = currentVersion.getDataset();
+        Long minor = currentVersion.getMinorVersionNumber();
+        if (Objects.equals(minor, 0L)) {
+            return dataset.getMostRecentMajorVersionReleaseDate();
+        }
+        Long major = currentVersion.getVersionNumber();
+        for (DatasetVersion version : dataset.getVersions()) {
+            if (Objects.equals(version.getVersionNumber(), major)
+                && Objects.equals(version.getMinorVersionNumber(), 0L)) {
+                return version.getReleaseTime();
+            }
+        }
+        return dataset.getMostRecentMajorVersionReleaseDate();
+    }
```

Comment:
I've added some clarifying comments, because I think, that it's better to not duplicate code from `getMostRecentMajorVersionReleaseDate()`. On the other hand, current method was written in order to circumvent some deficiencies of `getMostRecentMajorVersionReleaseDate`.

Reply:
Ok, somehow I was not aware that `getMostRecentMajorVersionReleaseDate` was used only in that single place. So I've followed your idea and ultimately removed `findReleaseDate`, having replaced it by corrected version of `getMostRecentMajorVersionReleaseDate` (which was also moved from `Dataset` to `IndexServiceBean`).